xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-test/xwiki-platform-ckeditor-test-docker/src/test/it/org/xwiki/ckeditor/test/ui/QuickActionsIT.java

```diff
@@ -0,0 +1,173 @@
+        textArea.sendKeys("hea");
+        qa.waitForQuickActionSelected("Heading 1");
+        textArea.sendKeys(Keys.ENTER);
+        qa.waitForQuickActionSubmitted();
+        
+        // Write some text
+        textArea.sendKeys(TEST_TEXT);
+        
+        assert textArea.getContent().contains("<h1>" + TEST_TEXT + "<br></h1>");
+        editPage.clickSaveAndView();
+    }
+    
+    @Test
+    @Order(2)
+    void heading2(TestUtils setup, TestReference testReference) throws Exception
+    {
+        WYSIWYGEditPage editPage = setup.gotoPage(testReference).editWYSIWYG();
+        CKEditor editor = new CKEditor("content").waitToLoad();
+        
+        RichTextAreaElement textArea = editor.getRichTextArea();
+
+        // Run the action
+        QuickActionsDropdown qa = editor.openQuickActionsDropdown();
+        textArea.sendKeys("hea");
+        qa.waitForQuickActionSelected("Heading 1");
+        textArea.sendKeys(Keys.DOWN);
+        qa.waitForQuickActionSelected("Heading 2");
+        textArea.sendKeys(Keys.ENTER);
+        qa.waitForQuickActionSubmitted();
+        
+        // Write some text
+        textArea.sendKeys(TEST_TEXT);
+        
+        assert textArea.getContent().contains("<h2>" + TEST_TEXT + "<br></h2>");
+        editPage.clickSaveAndView();
+    }
+    
+    @Test
+    @Order(3)
+    void heading3(TestUtils setup, TestReference testReference) throws Exception
+    {
+        WYSIWYGEditPage editPage = setup.gotoPage(testReference).editWYSIWYG();
+        CKEditor editor = new CKEditor("content").waitToLoad();
+        
+        RichTextAreaElement textArea = editor.getRichTextArea();
+
+        // Run the action
+        QuickActionsDropdown qa = editor.openQuickActionsDropdown();
+        textArea.sendKeys("hea");
+        qa.waitForQuickActionSelected("Heading 1");
+        textArea.sendKeys(Keys.DOWN, Keys.DOWN);
+        qa.waitForQuickActionSelected("Heading 3");
+        textArea.sendKeys(Keys.ENTER);
+        qa.waitForQuickActionSubmitted();
+        
+        // Write some text
+        textArea.sendKeys(TEST_TEXT);
+        
+        assert textArea.getContent().contains("<h3>" + TEST_TEXT + "<br></h3>");
+        editPage.clickSaveAndView();
+    }
+    
+    @Test
+    @Order(4)
+    void paragraph(TestUtils setup, TestReference testReference) throws Exception
+    {
+        WYSIWYGEditPage editPage = setup.gotoPage(testReference).editWYSIWYG();
+        CKEditor editor = new CKEditor("content").waitToLoad();
+        
+        RichTextAreaElement textArea = editor.getRichTextArea();
+
+        // Switch to another style
+        QuickActionsDropdown qa = editor.openQuickActionsDropdown();
+        textArea.sendKeys("hea");
+        qa.waitForQuickActionSelected("Heading 1");
+        textArea.sendKeys(Keys.ENTER);
+        qa.waitForQuickActionSubmitted();
+        
+        // Write some text
+        textArea.sendKeys(TEST_TEXT);
```

Comment:
You could assert here the HTML and drop the first heading test. It's fine to have a "story", i.e. to test multiple things in functional tests (compared to unit tests, which are less expensive).